hoss/coordinate_utilities.py

```diff
@@ -179,42 +182,220 @@ def get_1D_dim_array_data_from_dimvalues(
+    lat_lon_array: np.ndarray, coordinate: VariableFromDmr
 ) -> np.ndarray:
     """
     Returns indices of a valid array without fill values if the fill
     value is provided. If it is not provided, we check for valid values
     for latitude and longitude
     """
+
     # get_attribute_value returns a value of type `str`
     coordinate_fill = coordinate.get_attribute_value('_FillValue')
     if coordinate_fill is not None:
-        is_not_fill = ~np.isclose(coordinate_row_col, float(coordinate_fill))
+        is_not_fill = ~np.isclose(lat_lon_array, float(coordinate_fill))
     else:
         # Creates an entire array of `True` values.
-        is_not_fill = np.ones_like(coordinate_row_col, dtype=bool)
+        is_not_fill = np.ones_like(lat_lon_array, dtype=bool)
 
     if coordinate.is_longitude():
-        valid_indices = np.where(
-            np.logical_and(
-                is_not_fill,
-                np.logical_and(
-                    coordinate_row_col >= -180.0, coordinate_row_col <= 360.0
-                ),
-            )
-        )[0]
+        valid_indices = np.logical_and(
+            is_not_fill,
+            np.logical_and(lat_lon_array >= -180.0, lat_lon_array <= 360.0),
+        )
     elif coordinate.is_latitude():
-        valid_indices = np.where(
-            np.logical_and(
-                is_not_fill,
-                np.logical_and(coordinate_row_col >= -90.0, coordinate_row_col <= 90.0),
-            )
-        )[0]
+        valid_indices = np.logical_and(
+            is_not_fill,
+            np.logical_and(lat_lon_array >= -90.0, lat_lon_array <= 90.0),
+        )
     else:
         valid_indices = np.empty((0, 0))
 
-    return valid_indices
+    return valid_indices  # throw an exception
+
+
+def get_row_col_geo_grid_points(
+    lat_arr: np.ndarray,
+    lon_arr: np.ndarray,
+    latitude_coordinate: VariableFromDmr,
+    longitude_coordinate: VariableFromDmr,
+    # ) -> tuple[dict, dict]:
+) -> tuple[list, list]:
+    """
+    This method is used to return two valid lat lon points from a 2D
+    coordinate dataset. It gets the row and column of the latitude and longitude
+    arrays to get two valid points. This does a check for fill values and
+    This method does not go down to the next row and col. if the selected row and
+    column all have fills, it will raise an exception in those cases.
+    """
+
+    geo_grid_indices_row, geo_grid_indices_col = get_row_col_valid_indices_in_dataset(
+        lat_arr, lon_arr, latitude_coordinate, longitude_coordinate
+    )
+
+    geo_grid_col_points = [
+        (lon_arr[row, col], lat_arr[row, col]) for row, col in geo_grid_indices_row
+    ]
+    geo_grid_row_points = [
+        (lon_arr[row, col], lat_arr[row, col]) for row, col in geo_grid_indices_col
+    ]
+
+    return geo_grid_row_points, geo_grid_col_points
+
+
+def get_x_y_values_from_geographic_points(points: list, crs: CRS) -> list[tuple]:
```

Comment:
I assume this is:
```
get_x_y_values_from_geographic_points(
    points: list(tuple), # list of data points as tuple in (lon, lat) order 
    crs: CRS             # CRS object from PyProj
) -> list[tuple]         # list of X-Y points as tuple in (X, Y) order
```

Reply:
You might have to check coordinate ordering.

Reply:
The ordering is in the next ticket.

Reply:
updated the comment - https://github.com/nasa/harmony-opendap-subsetter/pull/18/commits/0d6c01b70791b9f4eff1d79fd069ecbcd03d6144

Reply:
Actually, my comment was in reference to a common confusion in proj transformations - which is the lat and which is the lon value, and then which is the x and which is the y value.  I think proj takes values in the order longitude, latitude - to define x, y.  I try to think in terms of longitude being the X coordinate in a geographic (lat/lon) projection.  This code seems to be using the order latitude, longitude to define x, y.  And then, if you want the order y, x - then your order of latitude, longitude might be correct, but the results would be reversed.  Generally, this can only be resolved by execution and verification - testing to ensure it is correct.  Which you may have implemented.  It is hard to resolve by analysis.